source/simulator/initial_conditions.cc

```diff
@@ -214,6 +91,55 @@ namespace aspect
 
         std::vector<types::global_dof_index> local_dof_indices (finite_element.dofs_per_cell);
 
+#if DEAL_II_VERSION_GTE(8,5,0)
+        const VectorFunctionFromScalarFunctionObject<dim, double> &advf_init_function =
+          (advf.is_temperature()?
+           VectorFunctionFromScalarFunctionObject<dim, double>(std_cxx11::bind(&InitialConditions::Interface<dim>::initial_temperature,
+                                                                               std::ref(*initial_conditions),
+                                                                               std_cxx11::_1),
+                                                               introspection.component_indices.temperature,
+                                                               introspection.n_components):
+           VectorFunctionFromScalarFunctionObject<dim, double>(std_cxx11::bind(&CompositionalInitialConditions::Interface<dim>::initial_composition,
+                                                                               std::ref(*compositional_initial_conditions),
+                                                                               std_cxx11::_1,
+                                                                               n-1),
+                                                               introspection.component_indices.compositional_fields[n-1],
+                                                               introspection.n_components));
+
+        const ComponentMask advf_mask =
+          (advf.is_temperature()?
+           introspection.component_masks.temperature:
+           introspection.component_masks.compositional_fields[n-1]);
+
+        VectorTools::interpolate(*mapping,
+                                 dof_handler,
+                                 advf_init_function,
+                                 initial_solution,
+                                 advf_mask);
+
+        if (n == 1 && parameters.normalized_fields.size()>0)
```

Comment:
In the other code path (for deal.II 8.4.0) the condition is written the other way `if (parameters.normalized_fields ... && n == 1)`. Not very important but for easier review: please order the conditions in this `if` the same way.